tests/baselines/reference/commonJSImportNestedClassTypeReference.js

```diff
@@ -0,0 +1,46 @@
+//// [tests/cases/conformance/salsa/commonJSImportNestedClassTypeReference.ts] ////
+
+//// [main.js]
+const { K } = require("./mod1");
+/** @param {K} k */
+function f(k) {
+    k.values()
+}
+
+//// [mod1.js]
+var NS = {}
+NS.K =class {
+    values() {
+    }
+}
+exports.K = NS.K;
+
+
+//// [mod1.js]
+"use strict";
+var NS = {};
+NS.K = /** @class */ (function () {
+    function K() {
+    }
+    K.prototype.values = function () {
+    };
+    return K;
+}());
+exports.K = NS.K;
+//// [main.js]
+"use strict";
+var K = require("./mod1").K;
+/** @param {K} k */
+function f(k) {
+    k.values();
+}
+
+
+//// [mod1.d.ts]
+export var K: {

```

Comment:
Hm - can we have an issue/update to get this to emit as
```ts
declare namespace NS {
    class _K {
      values(): void;
    }
    export {_K as K};
}
import _K = NS.K;
export {_K as K};
```
? That'd be more precise, and is reasonable enough given the (alias) symbol shape of the input, IMO. I imagine the serializer just isn't handling the NS nesting well right now, and it is falling back to a `var`-based emit, even though it all uses alias symbols.

Reply:
#40479